lib/address_composer.rb

```diff
@@ -25,12 +25,12 @@ def self.compose(components)
   attr_accessor :components
 
   def initialize(components)
-    self.components = components
-  end
+    self.components = components.dup
```

Comment:
I don't think the gem is responsible of knowing how to make a copy of the `components` receive. 
I would say there are two options here:

1. The code that uses the gem pass the shallow copy on initialization
2. We release a new version of the gem that doesn't touch the `components` passed and returns a new object at the end of the process